react/src/components/MaintenanceSettingList.tsx

```diff
@@ -0,0 +1,140 @@
+import { useSuspendedBackendaiClient } from '../hooks';
+import {
+  CLOSING_DURATION,
+  useSetBAINotification,
+} from '../hooks/useBAINotification';
+import SettingList, { SettingGroup } from './SettingList';
+import { RedoOutlined } from '@ant-design/icons';
+import { Button } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+const MaintenanceSettingList = () => {
+  const [isRecalculating, setIsRecalculating] = useState(false);
+  const [isRescanning, setIsRescanning] = useState(false);
+
+  const { t } = useTranslation();
+  const { upsertNotification } = useSetBAINotification();
+  const baiClient = useSuspendedBackendaiClient();
+
+  const recalculateUsage = () => {
+    setIsRecalculating(true);
+
+    upsertNotification({
+      message: t('maintenance.RecalculateUsage'),
+      description: t('maintenance.Recalculating'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+        promise: baiClient.maintenance
+          .recalculate_usage()
+          .finally(() => setIsRecalculating(false)),
+        statusDescriptions: {
+          pending: t('maintenance.Recalculating'),
+          resolved: t('maintenance.RecalculationFinished'),
+          rejected: t('maintenance.RecalculationFailed'),
+        },
+      },
+    });
+  };
+
+  const rescanImages = () => {
+    setIsRescanning(true);
+    const notiKey = upsertNotification({
+      message: t('maintenance.RescanImages'),
+      description: t('maintenance.RescanImageScanning'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+      },
+    });
+    // If values can be passed through resolve, please refactor the following function using promises
+    baiClient.maintenance
+      .rescan_images()
+      .then(({ rescan_images }) => {
+        if (rescan_images.ok) {
+          upsertNotification({
+            key: notiKey,
+            backgroundTask: {
+              status: 'pending',
+              percent: 0,
+              taskId: rescan_images.task_id,
+              statusDescriptions: {
+                pending: t('maintenance.RescanImageScanning'),
+                resolved: t('maintenance.RescanImageFinished'),
+                rejected: t('maintenance.RescanFailed'),
+              },
```

Comment:
```suggestion
              },
              to: '/environment',
              toText: 'Go to environment page'
```
How about setting `to` and `toText` props? This is just an example, so feel free to set the toText as you like. Don't forget to apply i18n.

Reply:
Yes, It's better to improve after adding new feature to BAINotification in separated issue.